### PR TITLE
Move mania stage hint to its own class

### PR DESCRIPTION
**Pull request by mcendu**

### CODE REVIEW

**Comment by peppy** [comment]
Is this a prereq for skinning support?

**Comment by mcendu** [comment]
Mostly yes

File: osu.Game.Rulesets.Mania/UI/Components/StageHint.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+using osu.Framework.Graphics.Containers;
+using osu.Game.Graphics;
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Mania.UI.Components
+{
+    public abstract class StageHint : CompositeDrawable, IHasAccentColour
+    {
+        public abstract Color4 AccentColour { get; set; }
```
Comment (mcendu) [inline osu.Game.Rulesets.Mania/UI/Components/StageHint.cs]: Change to virtual as legacy stage hints wouldn't be colored.

File: osu.Game.Rulesets.Mania/UI/Components/StageHint.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics.Containers;
+using osu.Game.Graphics;
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Mania.UI.Components
+{
+    public abstract class StageHint : CompositeDrawable, IHasAccentColour
```
Comment (peppy) [inline osu.Game.Rulesets.Mania/UI/Components/StageHint.cs]: not sure this base implementation need exist if it's this empty. if accent colour is the only usage, you can just null-cast is to pass in the colour:

```csharp
if (hitTarget is IhasAccentColour colouredHitTarget)
	colouredHitTarget.AccentColour = AccentColour;
```

**Review by peppy — CHANGES_REQUESTED** [review]

**Comment by peppy** [comment]
Not sure about the "stage hint" terminology. Where does this come from?

**Comment by GSculerlor** [comment]
If this PR is targeting for skin support, then it should be referring to `mania-stage-hint` (source from [wiki](https://osu.ppy.sh/help/wiki/Skinning/osu%21mania)). But on stable, `mania-stage-hint` is for full-width playfield, not per column

File: osu.Game.Rulesets.Mania/UI/Components/ColumnHitObjectArea.cs [hunk 3]
```diff
@@ -78,21 +60,86 @@ public Color4 AccentColour
...
+
+            [BackgroundDependencyLoader]
+            private void load(IScrollingInfo scrollingInfo)
+            {
+                direction.BindTo(scrollingInfo.Direction);
+                direction.BindValueChanged(dir =>
+                {
+                    Anchor anchor = dir.NewValue == ScrollingDirection.Up ? Anchor.TopLeft : Anchor.BottomLeft;
+
+                    hitTargetBar.Anchor = hitTargetBar.Origin = anchor;
```
Comment (peppy) [inline osu.Game.Rulesets.Mania/UI/Components/ColumnHitObjectArea.cs]: not sure doing the full bindable flow here is required (could just pass on the anchor from self to children) but should be fine for now.

**Review by peppy — APPROVED** [review]

**Pull request merged**